src/EditorFeatures/Core.Wpf/InlineRename/InlineRenameSession.cs

```diff
@@ -498,11 +509,15 @@ private void QueueApplyReplacements()
                     TaskContinuationOptions.OnlyOnRanToCompletion,
                     TaskScheduler.Default)
                 .Unwrap()
-                .SafeContinueWith(
-                    t => ApplyReplacements(t.Result.replacementInfo, t.Result.mergeResult, _conflictResolutionTaskCancellationSource.Token),
+                .SafeContinueWithFromAsync(
+                    async t =>
+                    {
+                        await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(alwaysYield: true, _conflictResolutionTaskCancellationSource.Token);
+                        ApplyReplacements(t.Result.replacementInfo, t.Result.mergeResult, _conflictResolutionTaskCancellationSource.Token);
+                    },
                     _conflictResolutionTaskCancellationSource.Token,
-                    TaskContinuationOptions.OnlyOnRanToCompletion,
-                    ForegroundTaskScheduler)
+                    TaskContinuationOptions.OnlyOnRanToCompletion | TaskContinuationOptions.ExecuteSynchronously,
```

Comment:
why ExecuteSynchronously?

Reply:
➡️ This pattern is the subject of the discussion in https://github.com/dotnet/roslyn/pull/28781#discussion_r208378244